### PR TITLE
fix(chat): add no models or application in marketplace (Issue #2448)

### PR DESCRIPTION
**Pull request by Derikyan**
**Description:**

Incorrect screen if to remove all models, applications from My workspace

What happened?
open Dial Marketplace -> switch to My workspace
remove all the applications and models (click on bookmark icon)
note that the screen shown info about search -> bug

Issues:

- Issue #2448 

**UI changes**

Снимок экрана 2024-10-28 в 00 23 51

**Checklist:**

- [x] the pull request name complies with [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/)
- [x] the pull request name starts with `fix(<scope>):`, `feat(<scope>):`, `feature(<scope>):`, `chore(<scope>):`, `hotfix(<scope>):` or `e2e(<scope>):`. If contains breaking changes then the pull request name must start with `fix(<scope>)!:`, `feat(<scope>)!:`, `feature(<scope>)!:`, `chore(<scope>)!:`, `hotfix(<scope>)!:` or `e2e(<scope>)!:` where `<scope>` is name of affected project: `chat`, `chat-e2e`, `overlay`, `shared`, `sandbox-overlay`, etc.
- [x] the pull request name ends with `(Issue #<TICKET_ID>)` (comma-separated list of issues)
- [x] I confirm that do not share any confidential information like API keys or any other secrets and private URLs

### CODE REVIEW

**Comment by Derikyan** [comment]
/deploy-review

>Environment URL: https://chat-ai-dial-chat-pr-2453.nightly-test.deltixhub.io
>E2E tests status: success

**Comment by Derikyan** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-2453.nightly-test.deltixhub.io
>E2E tests status: success

File: apps/chat/src/components/Marketplace/TabRenderer.tsx [hunk 1]
```diff
@@ -309,10 +318,25 @@ export const TabRenderer = ({ screenState }: TabRendererProps) => {
             </>
           ) : (
             <div className="flex grow flex-col items-center justify-center">
-              <NoResultsFound iconSize={100} className="gap-5 text-lg" />
-              <span className="mt-4 text-sm">
-                {t("Sorry, we couldn't find any results for your search.")}
-              </span>
+              {selectedTab === MarketplaceTabs.MY_APPLICATIONS &&
+              !searchTerm.length ? (
```
Comment (IlyaBondar) [inline apps/chat/src/components/Marketplace/TabRenderer.tsx]: we should check also that there are no filters and "No models" can be teoretically for DIAL Marketplace (some error diuring models uploading)

File: apps/chat/src/components/Marketplace/TabRenderer.tsx [hunk 2]
```diff
@@ -145,16 +146,24 @@ export const TabRenderer = ({ screenState }: TabRendererProps) => {
         : filteredEntities;
 
     const shouldSuggest =
-      selectedTab === MarketplaceTabs.MY_APPLICATIONS && !entitiesForTab.length;
+      selectedTab === MarketplaceTabs.MY_APPLICATIONS &&
+      !entitiesForTab.length &&
+      searchTerm.length;
```
Comment (IlyaBondar) [inline apps/chat/src/components/Marketplace/TabRenderer.tsx]: or if any other filters (Type or topics)

File: apps/chat/src/components/Marketplace/TabRenderer.tsx [hunk 3]
```diff
@@ -145,16 +146,24 @@ export const TabRenderer = ({ screenState }: TabRendererProps) => {
...
       ({ entities }) => orderBy(entities, 'version', 'desc')[0],
     );
 
-    setSuggestedResults(shouldSuggest ? orderedEntities : null);
+    if (shouldSuggest) {
+      setSuggestedResults(orderedEntities);
+      return [];
+    } else {
+      setSuggestedResults([]);
+    }
```
Comment (IlyaBondar) [inline apps/chat/src/components/Marketplace/TabRenderer.tsx:278]: you shouldn't calculate `groupedEntities` and `orderedEntities` if you aren't going to use it
```suggestion
    if (shouldSuggest) {
      setSuggestedResults(orderedEntities);
      return [];
    }
    setSuggestedResults([]);
    const groupedEntities = groupModelsAndSaveOrder(
      shouldSuggest ? filteredEntities : entitiesForTab,
    );

    const orderedEntities = groupedEntities.map(
      ({ entities }) => orderBy(entities, 'version', 'desc')[0],
    );
    return orderedEntities ;
```

Comment (Alexander-Kezik) [reply apps/chat/src/components/Marketplace/TabRenderer.tsx:278]: `groupEntities` used to set suggested results as I see

**Comment by Derikyan** [comment]
/deploy-review
>- **Deployment failed**: [failed](https://gitlab.deltixhub.com/Clients/deltix/nightly-test/dial/review/-/pipelines/531611)

File: apps/chat/src/components/Marketplace/TabRenderer.tsx [hunk 4]
```diff
@@ -271,6 +288,25 @@ export const TabRenderer = ({ screenState }: TabRendererProps) => {
...
+
+  const NoResultsFoundMessage = () => (
+    <div className="flex grow flex-col items-center justify-center">
+      <NoResultsFound iconSize={100} className="gap-5 text-lg" />
+      <span className="mt-4 text-sm font-normal">
+        {t("Sorry, we couldn't find any results for your search.")}
+      </span>
+    </div>
+  );
+
```
Comment (Alexander-Kezik) [inline apps/chat/src/components/Marketplace/TabRenderer.tsx]: move this components out of the component
Also we can create a component with `children`, `header`, `description` props. They look pretty same

File: apps/chat/src/components/Marketplace/TabRenderer.tsx [hunk 5]
```diff
@@ -293,51 +330,41 @@ export const TabRenderer = ({ screenState }: TabRendererProps) => {
...
           {selectedTab === MarketplaceTabs.MY_APPLICATIONS &&
-          suggestedResults.length ? (
-            <>
-              <div className="mb-8 flex items-center gap-1">
-                <Magnifier height={32} width={32} className="text-secondary" />
-                <span className="text-base">
-                  {t(
-                    'No results found in My workspace. Look at suggested results from DIAL Marketplace.',
-                  )}
+          isAllFiltersEmpty ? (
```
Comment (Alexander-Kezik) [inline apps/chat/src/components/Marketplace/TabRenderer.tsx]: it's hard to read such conditions. Maybe we can create some View component and return JSX depending on some conditions?

File: apps/chat/src/components/Marketplace/TabRenderer.tsx (hunk 4 above)
Comment (Derikyan) [reply apps/chat/src/components/Marketplace/TabRenderer.tsx]: What to do with "t" ?

Comment (Derikyan) [reply apps/chat/src/components/Marketplace/TabRenderer.tsx]: For each make its own 
const { t } = useTranslation(Translation.Marketplace); 
?

**Comment by IlyaBondar** [comment]
/deploy-review
>- **Deployment failed**: [failed](https://gitlab.deltixhub.com/Clients/deltix/nightly-test/dial/review/-/pipelines/531641)

File: apps/chat/src/components/Marketplace/TabRenderer.tsx [hunk 6]
```diff
@@ -119,6 +120,16 @@ export const TabRenderer = ({ screenState }: TabRendererProps) => {
   }>();
   const [detailsModelReference, setDetailsModelReference] = useState<string>();
 
+  const isSomeFilterNotEmpty =
+    searchTerm.length ||
+    selectedFilters[FilterTypes.ENTITY_TYPE].length ||
+    selectedFilters[FilterTypes.TOPICS].length;
+
+  const isAllFiltersEmpty =
```
Comment (IlyaBondar) [inline apps/chat/src/components/Marketplace/TabRenderer.tsx]:
```suggestion
  const areAllFiltersEmpty =
```

File: apps/chat/src/components/Marketplace/TabRenderer.tsx (hunk 4 above)
Comment (IlyaBondar) [reply apps/chat/src/components/Marketplace/TabRenderer.tsx]: At least these components must be out of this component

**Comment by Derikyan** [comment]
/deploy-review
>- **Deployment failed**: [failed](https://gitlab.deltixhub.com/Clients/deltix/nightly-test/dial/review/-/pipelines/531668)
>Environment URL: https://chat-ai-dial-chat-pr-2453.nightly-test.deltixhub.io
>E2E tests status: failed

File: apps/chat/src/components/Marketplace/ResultsView.tsx [hunk 7]
```diff
@@ -0,0 +1,119 @@
...
+    >
+      <IconMessage2 size={100} className="stroke-[0.2]" />
+    </NoAgentsFound>
+  );
+
+  const NoResultsFoundMessage = () => (
+    <NoAgentsFound descr={noResultsText}>
+      <NoResultsFound iconSize={100} className="gap-5 text-lg" />
+    </NoAgentsFound>
+  );
```
Comment (IlyaBondar) [inline apps/chat/src/components/Marketplace/ResultsView.tsx]: move these components out of this component (will be created each time)

**Comment by IlyaBondar** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-2453.nightly-test.deltixhub.io
>E2E tests status: success

**Comment by Derikyan** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-2453.nightly-test.deltixhub.io
>E2E tests status: failed

**Comment by IlyaBondar** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-2453.nightly-test.deltixhub.io
>E2E tests status: success

**Review by IlyaBondar — APPROVED** [review]

**Pull request merged**